source/SpinalHDL/Data types/bits.rst

```diff
@@ -41,7 +41,12 @@ The syntax to declare a bit vector is as follows: (everything between [] is opti
 .. code-block:: scala
 
    // Declaration
-   val myBits  = Bits()     // the size is inferred     
+   val myBits  = Bits()     // the size is inferred through later assignment
```

Comment:
I should have been clearer: The width is inferred from the assignments (actually the biggest known width) - so 6 bit width is inferred in your example.
But you try to assign a 4 bit value to a (inferred size) 6 bit value, which generates an error.

Reply:
So we ended up as:
```
   // Declaration
   val myBits  = Bits()     // the size is inferred from the widest assignment
   // ....
   // .resized needed to prevent WIDTH MISMATCH error as natural width does not match inferred size
   myBits := B("1010").resized  // auto-widen Bits(4 bits) to Bits(6 bits)
   when(condxMaybe) {
     myBits := B("110000")      // Bits(6 bits) would be inferred for myBits, this is the widest assignment
   }
```

This does help clarify the inference of hardware matter from the inference of Scala matters, which is the usual use of the term.